pyuvdata/uvdata/uvh5.py

```diff
@@ -445,31 +443,61 @@ def _read_header(
                 header["eq_coeffs_convention"][()]
             ).decode("utf8")
 
-        # We've added a new keyword that did not exist before, so check to see if it
-        # is in the header, and if not, mark the data set as being "regular" (i.e.,
-        # not a flexible spectral window setup).
+        # We've added a few new keywords that did not exist before, so check to see if
+        # it is in the header, and if not, mark the data set as being "regular" (e.g.,
+        # not a flexible spectral window setup, single source only).
         if "flex_spw" in header:
             if bool(header["flex_spw"][()]):
                 self._set_flex_spw()
         if "flex_spw_id_array" in header:
-            self.flex_spw_id_array = header["flex_spw_id_array"][()]
+            self.flex_spw_id_array = header["flex_spw_id_array"][:]
+        if "multi_object" in header:
+            if bool(header["multi_object"][()]):
+                self._set_phased()
+                self._set_multi_object(preserve_source_info=False)
+        if "Nobjects" in header:
+            self.Nobjects = int(header["Nobjects"][()])
```

Comment:
I'll let @plaplant decide if this should be integrated with the versioning he's planning in any way.

Reply:
After confirming w/ @plaplant, I've gone ahead and added the UVH5 versioning (v1.1!) in 52d2bd9d (and Paul will hopefully review it later this week).